skyportal/plot.py

```diff
@@ -818,17 +762,23 @@ def spectroscopy_plot(obj_id, user, spec_id=None):
         spec_labels.append(label)
 
     toggle = CheckboxWithLegendGroup(
-        labels=spec_labels,
+        labels=[
+            f'{s.instrument.telescope.nickname}/{s.instrument.name} ({s.observed_at.date().isoformat()})'
+            for s in spectra
+        ],
```

Comment:
this should be `spec_labels` as before, as the order of `spectra` is not the same as the order of `split`, so this results in the spectra being mislabeled

Reply:
Done